x-pack/plugin/identity-provider/src/internalClusterTest/java/org/elasticsearch/xpack/idp/action/SamlIdentityProviderTests.java

```diff
@@ -412,6 +459,28 @@ private void registerApplicationPrivileges(Map<String, Set<String>> privileges)
         assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
     }
 
+    private void createRole(String roleName, String roleBody, RequestOptions options) throws IOException {
+        Request req = new Request("PUT", "/_security/role/" + roleName);
+        req.setJsonEntity(roleBody);
+        req.setOptions(options);
+        getRestClient().performRequest(req);
+    }
+
+    private void createUser(String userName, SecureString password, String roleName, RequestOptions options) throws IOException {
+        final Request req = new Request("PUT", "/_security/user/" + userName);
+        final String body = """
+            {
+              "username": "%s",
+              "full_name": "Test User (%s)",
+              "password": "%s",
+              "roles": [ "%s" ]
+            }
+            """.formatted(userName, getTestName(), password, roleName);
+        req.setJsonEntity(body);
+        req.setOptions(options);
+        getRestClient().performRequest(req);
+    }
```

Comment:
Why not use `TestSecurityClient` here?

Reply:
Mostly to avoid dependencies on the `security` project.
That may not be important, since the IdP test already have that dependency, but the intent was to avoid such dependencies where possible.